public/app/features/search/state/SearchStateManager.ts

```diff
@@ -81,8 +81,11 @@ export class SearchStateManager extends StateManagerBase<SearchState> {
       sort: this.state.sort,
     });
 
-    // issue new search query
-    this.doSearchWithDebounce();
+    // Prevent searching when user is only clearing the input.
+    // We don't show these results anyway
+    if (this.hasSearchFilters()) {
```

Comment:
i was about to ask a whole bunch of questions about whether this relied on the previous `setState` on line 72, and given react's state updates are asynchronous do we think this is gonna cause problems, etc.

then i realised this isn't a react component, it's just a regular class that looks like a react class component because of the `setState` 😂